### ISSUE TITLE
Incorrect length for extension data

### ISSUE THREAD

**Issue by drur** [kriszyp/msgpackr#12]
Hello. I just noticed, when packing extension data with more than 255 bytes, the length written to the target buffer is incorrect.
If I'm not mistaken, the error is here, where it should be `target[position++] = length >> 8` instead:
https://github.com/kriszyp/msgpackr/blob/751ff8526309f9906848a3276834a0320b3f3192/pack.js#L707

## Example
```js
const assert = require('assert');
const {pack, unpack, addExtension} = require('msgpackr');

class TestClass {}

addExtension({
	Class: TestClass,
	type: 0x01,
	pack() {
		return Buffer.alloc(256);
	},
	unpack(data) {
		// here data.length equals to 0
		assert(data.length == 256);
	}
});
unpack(pack(new TestClass()));
```

**Comment by kriszyp** [comment]
Fixed in 1.2.0, thanks for the report.

**Comment by drur** [comment]
It works now, thanks for the quick fix!
